src/image-copy/azext_imagecopy/custom.py

```diff
@@ -19,14 +21,24 @@
 # pylint: disable=too-many-branches
 def imagecopy(cmd, source_resource_group_name, source_object_name, target_location,
               target_resource_group_name, temporary_resource_group_name='image-copy-rg',
-              source_type='image', cleanup='false', parallel_degree=-1, tags=None, target_name=None,
+              source_type='image', cleanup=None, parallel_degree=-1, tags=None, target_name=None,
               target_subscription=None, export_as_snapshot='false', timeout=3600):
+    # If --cleanup is set, forbid using an existing temporary resource group name.
+    # It is dangerous to clean up an existing resource group.
+    if cleanup:
```

Comment:
There's a person that also had an issue when using the same source & target resource group name with cleanup=true. That essentially deleted the source group which included a bunch of other VMs.
Another check I had in mind is not allowing same source/target (on the same subscription) when cleaning up. But that might be for another PR...

Reply:
Do you mean temporary resource group and target resource group?

Reply:
sorry, I confused it a bit. 
I meant the source and the temporary one. There was an incident where a user gave the same same name for the temp one as the source and the process deleted many production resources.

Reply:
Source resource group is an existing resource group, so this problem can be covered by this piece of code.